File: Services/UnionService/UnionService.Application/UseCases/CompanyUseCases/AddCompanyWorker/AddCompanyWorkerCommand.cs
using MediatR;
using UnionService.Domain.Enums;

namespace UnionService.Application.UseCases
{
    public record AddCompanyWorkerCommand(
        string? username,
        int? companyId,
        string? memberUsername,
        CompanyRole? role,
        decimal? salary,
        int? workHours, 
        int? workDays
        ) : IRequest<Unit>;
}

File: FrontendApp/src/app/models/commands/unionservice/add-company-worker.command.ts
import { CompanyRole } from "../../../enums/company-role.enum";

export interface AddCompanyWorkerCommand {
  username: string | null;
  companyId: number | null;
  memberUsername: string | null;
  role: CompanyRole | null;
  salary: number | null;
  workHours: number | null;
  workDays: number | null;
}
